crates/service-protocol/src/discovery.rs

```diff
@@ -410,6 +410,17 @@ impl ServiceDiscovery {
                     .into_parts();
 
                 if !parts.status.is_success() {
+                    let body_message = body
+                        .collect()
+                        .await
+                        .map(|b| {
+                            String::from_utf8_lossy(b.to_bytes().to_vec().as_slice()).to_string()
+                        })
+                        .unwrap_or_else(|err| format!("Failed to read body {}", err));
+                    warn!(
+                        "Bad status code '{}' when discovering deployment at address '{}'. Response : {:?}",
+                        parts.status, address, body_message
+                    );
```

Comment:
How about you add the `body_message` to the `DiscoveryError::BadStatusCode`, similar to how `DiscoveryError::BadResponse` is defined? This will make sure the error message is printed once with the returned error, containing the headers too.

Reply:
thanks for the suggestion, should be ok now